src/main/java/org/sonatype/repository/vgo/internal/hosted/VgoHostedFacetImpl.java

```diff
@@ -0,0 +1,239 @@
+{
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private final VgoDataAccess vgoDataAccess;
+
+  @Override
+  protected void doInit(final Configuration configuration) throws Exception {
+    super.doInit(configuration);
+  }
+
+  @Inject
+  public VgoHostedFacetImpl(final VgoDataAccess vgoDataAccess) {
+    this.vgoDataAccess = checkNotNull(vgoDataAccess);
+  }
+
+  @Override
+  @Nullable
+  @Transactional
+  public Content getInfo(final String path,
+                         final VgoAttributes vgoAttributes) {
+    checkNotNull(path);
+    String newPath = path.replaceAll("\\.info", "\\.zip");
+
+    StorageTx tx = UnitOfWork.currentTx();
+
+    Asset asset = vgoDataAccess.findAsset(tx, tx.findBucket(getRepository()), newPath);
+    if (asset == null) {
+      return null;
+    }
+
+    StreamPayload streamPayload = new StreamPayload(
+        () -> doGetInfo(asset, vgoAttributes),
+        UNKNOWN_SIZE,
+        ContentTypes.APPLICATION_JSON);
+    return new Content(streamPayload);
+  }
+
+  @Transactional
+  @Override
+  public Content getList(final String module) {
+    checkNotNull(module);
+
+    StorageTx tx = UnitOfWork.currentTx();
+
+    Iterable<Asset> assetsForModule = vgoDataAccess.findPackageAssetsForModule(tx, getRepository(), module);
+
+    String listOfVersions = StreamSupport.stream(assetsForModule.spliterator(), false)
+        .map(asset -> asset.name())
+        .map(name -> name.split("/@v/")[1])
+        .map((name -> name.replaceAll(".zip", "")))
+        .collect(Collectors.joining("\n"));
+
+    return new Content(
+      new StreamPayload(
+          new InputStreamSupplier() {
+            @Nonnull
+            @Override
+            public InputStream get() {
+              return new ByteArrayInputStream(listOfVersions.getBytes());
+            }
+          },
+          UNKNOWN_SIZE,
+          ContentTypes.TEXT_PLAIN
+      )
+    );
+  }
+
+  private InputStream doGetInfo(final Asset asset, final VgoAttributes vgoAttributes) {
+    VgoInfo vgoInfo = new VgoInfo(vgoAttributes.getVersion(), asset.blobCreated().toString());
+    try {
+      String info = MAPPER.writeValueAsString(vgoInfo);
+      return new ByteArrayInputStream(info.getBytes());
+    }
+    catch (JsonProcessingException e) {
+      log.warn(String.format("Unable to convert %s to json", vgoInfo.toString()));
+    }
+    return EmptyInputStream.INSTANCE;
+  }
+
+  @Nullable
+  @TransactionalTouchBlob
+  @Override
+  public Content getPackage(final String path) {
+    return doGet(path);
+  }
+
+  @Nullable
+  @TransactionalTouchBlob
+  @Override
+  public Content getMod(final String path) {
+    return doGet(path);
+  }
+
+  private Content doGet(final String path) {
+    checkNotNull(path);
+    StorageTx tx = UnitOfWork.currentTx();
+
+    Asset asset = vgoDataAccess.findAsset(tx, tx.findBucket(getRepository()), path);
+    if (asset == null) {
+      return null;
+    }
+
+    if (asset.markAsDownloaded()) {
```

Comment:
This should be handled by the AssetDownloadHandler now

Reply:
https://github.com/sonatype-nexus-community/nexus-repository-vgo/pull/5/commits/372baece1a6c2bbbe29e524f609bb5bbbfb16c15